### PR TITLE
[Reviewer: Matt] Use LZ4 for compression

### PR DESCRIPTION
**Pull request by rkd-msw**
I'll update the timestamp before I merge this, but I'm doing that as late as possible to avoid spurious merge conflicts (e.g. with @chris-elford-metaswitch 's story).

### CODE REVIEW

File: clearwater_sas_resource_bundle.yaml [hunk 1]
```diff
@@ -1801,11 +1801,14 @@ compression_profiles:
     sip:
         # This must match PROFILE_SIP defined in https://github.com/Metaswitch/cpp-common/blob/master/include/sasevent.h
         dictionary: "PCMUACKrportSUBSCRIBEP-Access-Network-Info: BYEusert=0 0 telephone-eventAccept: transportREGISTERSubscription-State: NOTIFYServer: m=audio RTP/AVP c=IN IP4 Expires: 200 OK\r\na=rtpmap:INVITETo: application/sdpVia: Content-Type: From: CSeq: Max-Forwards: Contact: Organization: Content-Length: Call-ID: ;tag=;branch=z9hG4bKSIP/2.0/UDP<sip:"
+        algorithm: "lz4"
```
Comment (mirw) [inline clearwater_sas_resource_bundle.yaml]: Convention elsewhere (e.g. "SIP" and "HTTP") seems to be not to use double-quotes, and to use capitals if that's the correct form.  I'm not sure what the right answer is, but worth getting the SAS team's input if you haven't already.

**Comment by mirw** [comment]
Looks good.  One comment.

File: clearwater_sas_resource_bundle.yaml (hunk 1 above)
Comment (rkd-msw) [inline clearwater_sas_resource_bundle.yaml]: This is the form used in the SAS integration guide (including double quotes and lowercase).

Comment (mirw) [inline clearwater_sas_resource_bundle.yaml]: I spoke to @AlexHockey and that was a bit arbitrary, but we don't feel too strongly.

**Comment by rkd-msw** [comment]
I've updated this PR to add no-dictionary LZ4 decompression for Diameter. Might be worth you re-reviewing, but I've checked that it's valid YAML, which I think is the main risk.

**Pull request merged**